### PR TITLE
Update UI for Settings and associated views

### PR DESCRIPTION
**Pull request by danielnordh**
Updates the Settings view and associated subviews.
Follows previous style patterns established in the onboarding PR.

image
(not all darkmode screens are shown here)

## Summary by CodeRabbit

- **New Features**
	- Introduced new views for managing channels, peers, and settings, enhancing user interaction with the Lightning Network.
	- Added functionality for displaying and managing recovery phrases securely.
	- Added a new `ChannelAddView` for adding channels and a `SeedView` for displaying recovery phrases.

- **Bug Fixes**
	- Improved error handling and user feedback mechanisms across various views.

- **Refactor**
	- Renamed and restructured several existing views and view models for clarity and improved organization.
	- Removed redundant properties related to network color management from multiple view models.
	- Updated layout and structure of existing views to enhance usability and accessibility.
	- Enhanced the `ChannelDetailView` and `SettingsView` for better user experience.
	- Removed outdated views such as `NodeIDView`, `ChannelsListView`, and others to streamline the interface. 

- **Documentation**
	- Updated previews for new views to assist in development and debugging.

### CODE REVIEW

File: LDKNodeMonday/Service/Lightning Service/LightningNodeService.swift [hunk 1]
```diff
@@ -24,7 +24,7 @@ class LightningNodeService {
         let storedNetworkString = try! keyService.getNetwork() ?? Network.signet.description
         let storedEsploraURL =
             try! keyService.getEsploraURL()
-        ?? EsploraServer.lqwd_signet.url
+            ?? EsploraServer.lqwd_signet.url
```
[the remark this answers is not in the document]
Comment (danielnordh) [reply LDKNodeMonday/Service/Lightning Service/LightningNodeService.swift:27]: Address in separate PR

File: LDKNodeMonday/View/Settings/Peers/DisconnectView.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            ToolbarItem(placement: .navigationBarTrailing) {
+                Button {
+                    viewModel.disconnect()
+                    if showingDisconnectViewErrorAlert == false {
+                        self.presentationMode.wrappedValue.dismiss()
+                    }
+                    if showingDisconnectViewErrorAlert == true {
+                        self.presentationMode.wrappedValue.dismiss()
+                    }
+                } label: {
```
[the remark this answers is not in the document]
Comment (danielnordh) [reply LDKNodeMonday/View/Settings/Peers/DisconnectView.swift]: Done

File: LDKNodeMonday/View/Settings/Danger/SeedView.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                            ? "checkmark" : "doc.on.doc"
+                    ) {
+                        UIPasteboard.general.string = viewModel.seed.mnemonic
+                        isCopied = true
+                        showCheckmark = true
+                        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
+                            isCopied = false
+                            showCheckmark = false
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (danielnordh) [reply LDKNodeMonday/View/Settings/Danger/SeedView.swift]: Done

File: LDKNodeMonday/View/Settings/Danger/SeedView.swift [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import SwiftUI
+
+struct SeedView: View {
+    @ObservedObject var viewModel: SeedViewModel
+    @State private var showAlert = false
+    @State private var showRecoveryPhrase = false
+    @State private var isCopied = false
+    @State private var showCheckmark = false
+    @State private var showingSeedViewErrorAlert = false
+
```
[the remark this answers is not in the document]
Comment (danielnordh) [reply LDKNodeMonday/View/Settings/Danger/SeedView.swift:19]: Maybe at a later point.

File: LDKNodeMonday/View/Settings/Peers/PeerDetailsView.swift [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        }.alert(
+            "Are you sure you want to disconnect from this peer?",
+            isPresented: $showDisconnectAlert
+        ) {
+            Button("Yes", role: .destructive) {
+                viewModel.disconnect()
+                
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
+                    self.presentationMode.wrappedValue.dismiss()
+                }
```
[the remark this answers is not in the document]
Comment (danielnordh) [reply LDKNodeMonday/View/Settings/Peers/PeerDetailsView.swift]: This requires making viewModel.disconnect() async. I tried quickly but requires more work so suggest waiting.

File: LDKNodeMonday/View/Settings/Peers/PeerView.swift [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+                        self.presentationMode.wrappedValue.dismiss()
+                    }
+                }
+
+                if showingPeerViewErrorAlert == false {
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
+                        self.presentationMode.wrappedValue.dismiss()
+                    }
+                }
+
```
[the remark this answers is not in the document]
Comment (danielnordh) [reply LDKNodeMonday/View/Settings/Peers/PeerView.swift:143]: Fixed

File: LDKNodeMonday/View/Settings/Channel/ChannelDetailView.swift [hunk 7]
```diff
@@ -14,79 +14,74 @@ struct ChannelDetailView: View {
...
-                }
-                .buttonBorderShape(.capsule)
-                .buttonStyle(.borderedProminent)
-                .tint(viewModel.networkColor)
-                .padding(.all)
-                .padding(.bottom, 100.0)
+                .padding(.top, 5)
+                .listRowSeparator(.hidden)
+                .listRowBackground(Color.clear)
+            }
```
[the remark this answers is not in the document]
Comment (danielnordh) [reply LDKNodeMonday/View/Settings/Channel/ChannelDetailView.swift:69]: Fixed

File: LDKNodeMonday/View/Settings/Channel/ChannelDetailView.swift [hunk 8]
```diff
@@ -101,12 +97,12 @@ struct ChannelDetailView: View {
...
-                    refreshFlag = true
                     if !showingChannelDetailViewErrorAlert {
-                        DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
-                            self.presentationMode.wrappedValue.dismiss()
-                        }
+                        refreshFlag = true
+                        isClosing = false
+                        self.presentationMode.wrappedValue.dismiss()
                     }
                 }
```
[the remark this answers is not in the document]
Comment (danielnordh) [reply LDKNodeMonday/View/Settings/Channel/ChannelDetailView.swift:107]: This code was CodeRabbit's suggestion.
Seems good enough for now.

File: LDKNodeMonday/View/Settings/Peers/PeerView.swift [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+            }
```
[the remark this answers is not in the document]
Comment (danielnordh) [reply LDKNodeMonday/View/Settings/Peers/PeerView.swift]: Don't want to move more into viewModel in anticipation of moving some functionality to LightningNodeService

**Review by reez — APPROVED** [review]
This looks great! 

Thanks for adding screenshots side by side and light+dark mode, always helps a bunch.

I like some of the details in here like the `ChannelDetailsFormatted` where you can use `isCopyable` in `ChannelDetailView`.

Thanks for formatting and removing some unneeded code now like `getColor()` etc.

Good stuff!

**Pull request merged**